### PR TITLE
Change base image from debian to ubuntu and upgrade python version

### PR DESCRIPTION
**Pull request by bipinKrishnan**
The PR updates the dockerfile with the following changes:

1. Change the docker image from Debian buster to Ubuntu 22.04
2. Upgrade Python from 3.8 to 3.10
3. Make use of `PyOpenGL` pip package instead of the apt package

### CODE REVIEW

**Comment by MahmoudAshraf97** [comment]
Hi, is this an ongoing task? if not what is the motive for these changes?

**Comment by bipinKrishnan** [comment]
Hi @MahmoudAshraf97, the changes are based on the trello card here:
https://trello.com/c/mrDtbTks

**Review by VedPatwardhan — APPROVED** [review]
lgtm! Let's get the PR merged and if it breaks anything on master we could revert it, given that there isn't a way to exhaustively test if there are failures. Being able to build a codespace is a good check to know that it doesn't cause any major issues, probably run tests for a couple of submodules in the codespace you created before getting the PR merged. Thanks @bipinKrishnan 🙂

**Pull request merged**
